src/py/flwr/common/differential_privacy.py

```diff
@@ -39,7 +39,8 @@ def get_norm(input_arrays: NDArrays) -> float:
 def add_gaussian_noise_inplace(input_arrays: NDArrays, std_dev: float) -> None:
     """Add Gaussian noise to each element of the input arrays."""
     for array in input_arrays:
-        array += np.random.normal(0, std_dev, array.shape)
+        noise = np.random.normal(0, std_dev, array.shape).astype(array.dtype)
```

Comment:
Instead of `astype`, would it be possible to specify the dtype in the random function?
```suggestion
        noise = np.random.normal(0, std_dev, array.shape, dtype=array.dtype)
```